services/datamanager/builtin/file_deletion_test.go

```diff
@@ -0,0 +1,188 @@
+package builtin
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/hex"
+	"fmt"
+	"math"
+	"os"
+	"sync/atomic"
+	"testing"
+
+	v1 "go.viam.com/api/app/datasync/v1"
+	"go.viam.com/test"
+
+	"go.viam.com/rdk/logging"
+	"go.viam.com/rdk/services/datamanager/datasync"
+)
+
+func TestFileDeletionUsageCheck(t *testing.T) {
+	tests := []struct {
+		name                 string
+		expectedFsCheckValue bool
+		testThreshold        float64
+		testCaptureDirRatio  float64
+		captureDirExists     bool
+	}{
+		{
+			name:                 "if not at file system capacity threshold, we should return false from deletion check",
+			expectedFsCheckValue: false,
+			testThreshold:        .99,
+			testCaptureDirRatio:  .99,
+			captureDirExists:     true,
+		},
+		{
+			name:                 "if at file system capacity threshold, we return true from deletion check",
+			expectedFsCheckValue: true,
+			testThreshold:        math.SmallestNonzeroFloat64,
+			testCaptureDirRatio:  math.SmallestNonzeroFloat64,
+			captureDirExists:     true,
+		},
+		{
+			name: "if at file system capacity threshold but not capture dir threshold," +
+				"we return false from deletion check",
+			expectedFsCheckValue: false,
+			testThreshold:        math.SmallestNonzeroFloat64,
+			testCaptureDirRatio:  1.0,
+			captureDirExists:     true,
+		},
+		{
+			name:                 "if capture dir does not exist, we should return false from deletion check",
+			expectedFsCheckValue: false,
+			testThreshold:        .95,
+			testCaptureDirRatio:  .5,
+			captureDirExists:     false,
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			var tempCaptureDir string
+			if tc.captureDirExists {
+				tempCaptureDir = t.TempDir()
+				// overwrite thresholds
+				fsThresholdToTriggerDeletion = tc.testThreshold
+				captureDirToFSUsageRatio = tc.testCaptureDirRatio
+				// write testing files
+				writeFiles(t, tempCaptureDir, false)
+			} else {
+				// make a random dir name to make sure it doesn't exist
+				randomName := make([]byte, 4)
+				_, err := rand.Read(randomName)
+				test.That(t, err, test.ShouldBeNil)
+				tempCaptureDir = hex.EncodeToString(randomName)
+			}
+
+			logger := logging.NewTestLogger(t)
+			willDelete, err := shouldDeleteBasedOnDiskUsage(context.Background(), tempCaptureDir, logger)
+			test.That(t, err, test.ShouldBeNil)
+			test.That(t, willDelete, test.ShouldEqual, tc.expectedFsCheckValue)
+		})
+	}
+}
+
+func TestFileDeletion(t *testing.T) {
+	tests := []struct {
+		name                      string
```

Comment:
Overall, I think the tests would be much clearer if instead of booleans, we just had the values of what's being added. For example, instead of a boolean markAllInProgress, []string filesMarkedInProgress. Instead of a boolean writeProgFiles, maybe numDotCaptureFilesToWrite and numDotProgFilesToWrite